newIDE/app/src/Export/GenericExporters/OnlineWebExport.js

```diff
@@ -38,62 +59,134 @@ export const WebProjectLink = ({ build, loading }: WebProjectLinkProps) => {
   const [showCopiedInfoBar, setShowCopiedInfoBar] = React.useState<boolean>(
     false
   );
+  const [isShareDialogOpen, setIsShareDialogOpen] = React.useState<boolean>(
+    false
+  );
 
   if (!build && !loading) return null;
   const buildPending = loading || (build && build.status !== 'complete');
-
-  const value = buildPending
-    ? 'Just a few seconds while we generate the link...'
-    : getBuildArtifactUrl(build, 's3Key') || '';
+  const buildUrl = buildPending ? null : getBuildArtifactUrl(build, 's3Key');
 
   const onOpen = () => {
-    if (buildPending) return;
-    Window.openExternalURL(value);
+    if (buildPending || !buildUrl) return;
+    Window.openExternalURL(buildUrl);
   };
 
   const onCopy = () => {
-    if (buildPending) return;
+    if (buildPending || !buildUrl) return;
     // TODO: use Clipboard.js, after it's been reworked to use this API and handle text.
-    navigator.clipboard.writeText(value);
+    navigator.clipboard.writeText(buildUrl);
     setShowCopiedInfoBar(true);
   };
 
+  const onShare = async () => {
+    if (buildPending || !buildUrl) return;
+
+    // We are on desktop
```

Comment:
Uber nitpicking: put the comment inside the branch + period at the end. Also we could be on something just not supporting share.
```suggestion
    // We are on desktop (or do not support sharing using the system dialog).
```